### ISSUE TITLE
Pruning lower dimensional entities causes issues with cell_data

### ISSUE THREAD

**Issue by uvilla** [nschloe/meshio#789]
@nschloe ,
Thank you again for this great tool!

I found a small glitch in how `cell_data` is handled within `mesh.prune()`

Consider the following examples (which uses `pygalmesh`).

```python
import pygalmesh
import meshio
import numpy as np


if __name__ == '__main__':

    # Create a simple "brick" domain with 3 stripes
    data = np.zeros((100,100,100), dtype=np.uint8)
    data[:,20:40,20:80] = 1
    data[:,40:60,20:80] = 2
    data[:,60:80,20:80] = 3
    h    = [.01, .01, .01]
        
    mesh = pygalmesh.generate_from_array(data, h, facet_distance=.01, cell_size=.1, verbose=True)    
    

    meshio.xdmf.write("before_pruning.xdmf", mesh)
    
    mesh.prune()
    meshio.xdmf.write("after_pruning.xdmf", mesh)
```

The expected output of this code is a brick mesh with three layers corresponding to material ids 1,2,3. 

The figure below shows that `cell_data` is written correctly if the mesh is not pruned (left).
However, if we prune the mesh first, then the written `cell_data` is jitter (right).

![Image 4-24-20 at 2 37 PM](https://user-images.githubusercontent.com/17551415/80250594-c5528980-8639-11ea-9076-ec93fc625c35.jpg)

**Comment by uvilla** [comment]
Related to the above question,

Can you explain what is the content of `cell_data`?

In particular, why does
```python
print(mesh.cell_data)
```

Shows that the `medit:ref` data is a list of two numpy arrays?

```bash
{'medit:ref': [array([3, 3, 2, ..., 1, 1, 1]), array([2, 3, 3, ..., 1, 1, 1])]}
```

I assume that one array is the cell labels. What is the meaning of the other one?

**Comment by nschloe** [comment]
> Shows that the medit:ref data is a list of two numpy arrays?

Because `mesh.cells` is a list of two cellblocks.

**Comment by nschloe** [comment]
What is going wrong exactly with the prune? One cannot tell from the screenshot alone. Perhaps the triangles are removed?

**Comment by uvilla** [comment]
In `cgal` when creating a mesh from a domain with multiple labels each cell inherits a label corresponding to the subdomain it belonged to. Points are assigned a label which depends on the labels of the cells that are incident to the points.

Therefore in the figure of the left, you can see that each tetrahedron has the correct label (you can see the three layers).

In the figure on the right, you can see that tetrahedron have the wrong labels.

My suspicion is that `mesh.prune` removes lower dimensionality cells (as supposed to), but does not correctly prune `mesh.cell_data`.

**Comment by uvilla** [comment]
Ok here's the problem.

Before `mesh.prune()` we have two cell-blocks: `[0]` for triangle, `[1]` for tetrahedra.

```python
    print(mesh.cells)
    print(mesh.cell_data["medit:ref"])
```

Output
```bash
[CellBlock(type='triangle', data=array([[549, 399, 548],
       [548, 399, 549],
       [447, 482, 307],
       ...,
       [698, 639, 692],
       [639, 231, 698],
       [698, 231, 639]])), CellBlock(type='tetra', data=array([[823,  50, 873, 838],
       [428, 183, 180, 660],
       [548, 399, 545, 549],
       ...,
       [276, 242,  19, 698],
       [811, 231, 639, 698],
       [252, 275,  19, 698]]))]
[array([3, 3, 2, ..., 1, 1, 1]), array([2, 3, 3, ..., 1, 1, 1])]
```

After calling `mesh.prune()` we should expect to have only one cell block: `0` for tetrahedra.
However, `mesh.cell_data` was not updated correctly (it still has two arrays instead of one)`.

```python
    mesh.prune()
    
    print(mesh.cells)
    print(mesh.cell_data["medit:ref"])
```

Output:
```bash
[CellBlock(type='tetra', data=array([[823,  50, 873, 838],
       [428, 183, 180, 660],
       [548, 399, 545, 549],
       ...,
       [276, 242,  19, 698],
       [811, 231, 639, 698],
       [252, 275,  19, 698]]))]
[array([3, 3, 2, ..., 1, 1, 1]), array([2, 3, 3, ..., 1, 1, 1])]
```

The correct output should be:
```bash
[CellBlock(type='tetra', data=array([[823,  50, 873, 838],
       [428, 183, 180, 660],
       [548, 399, 545, 549],
       ...,
       [276, 242,  19, 698],
       [811, 231, 639, 698],
       [252, 275,  19, 698]]))]
[array([2, 3, 3, ..., 1, 1, 1])]
```

**Comment by uvilla** [comment]
Ok,

I confirm that the code below gives the correct result:

```python
import pygalmesh
import meshio
import numpy as np


if __name__ == '__main__':

    # Create a simple "brick" domain with 3 stripes
    data = np.zeros((100,100,100), dtype=np.uint8)
    data[:,20:40,20:80] = 1
    data[:,40:60,20:80] = 2
    data[:,60:80,20:80] = 3
    h    = [.01, .01, .01]
    
    print(data[:,:,5])
    
    mesh = pygalmesh.generate_from_array(data, h, facet_distance=.01, cell_size=.1, verbose=True)    
    

    meshio.xdmf.write("before_pruning.xdmf", mesh)
        
    mesh.prune()
    mesh.cell_data["medit:ref"].pop(0)
        
    print(mesh)
    meshio.xdmf.write("after_pruning.xdmf", mesh)
```

So I believe that the issue is [here](https://github.com/nschloe/meshio/blob/31fb15b23bdc6b3e58680208e3a9cb4b4a02dc1b/meshio/_mesh.py#L83)

```
new_cell_data[name] += data
```

It is concatenating all `data` rather than `data` of the tetrahedral block only.

**Comment by uvilla** [comment]
Got it:

Block 0: triangles
Block 1: tetrahedra

> > Shows that the medit:ref data is a list of two numpy arrays?
> 
> Because `mesh.cells` is a list of two cellblocks.

**Comment by nschloe** [comment]
> The correct output should be:

Want to submit a PR?

**Comment by nschloe** [comment]
Fixed by #790.
